app/controllers/api/v1/courses_controller.rb

```diff
@@ -1,9 +1,19 @@
 module Api
   module V1
     class CoursesController < ApplicationController
+      include NextLinkHeader
+
       def index
-        @courses = Course.includes(:sites, :provider, :site_statuses, :subjects)
-        paginate json: @courses
+        per_page = params[:per_page] || 100
+        changed_since = params[:changed_since]
+
+        @courses = Course.includes(:sites, :provider, :site_statuses, :subjects).changed_since(changed_since).limit(per_page)
+
+        next_link_header("from_course_id", @courses.last, changed_since, per_page)
+
+        render json: @courses
+      rescue ActiveRecord::StatementInvalid
```

Comment:
@misaka is there a clean way to DRY this out, since it's also present in the provider controller?

Reply:
Not sure there's much benefit to DRYing out two instances of a one-liner...

Reply:
Same, I agree with the [three strikes rule](http://wiki.c2.com/?ThreeStrikesAndYouRefactor), but there will be another I think... more thinking ahead.

Reply:
You could put it into a `before_action` hook. It might make sense to pull this param out and store the changed_since value in `@changed_since`, using `nil` as a signal that it's not set. Similar to how we might pull out an `id` value. Also, it would be less ambiguous than catching any and all `ActiveRecord::StatementInvalid` exceptions.

I'd wait to see what comes out of @dankmitchell's work to look at the parsing of `changed_since` before DRYing this up, though.

Reply:
Sensible suggestion 👍

Reply:
I'll revisit this once this PR is over the line.